### PR TITLE
Cleanup welcome screen recent databases view

### LINKED ISSUE #4710: Feature request : delete priviouly opened database entry’s
## Summary
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Provide a brief overview of what the new feature is all about )

When KeepassXC starts it display a list with the last opened databases . 

The idea is to right click on one entry to delete the information. 

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )

Background : sometimes network locations changes where my database is saved and I want to clean the interface of old entry’s .

### PR DESCRIPTION
**Pull request by JMcKiern**
While a database in the "Recent databases" on the welcome screen is selected, pressing the "Delete" key will now cause that database to be removed from the recent databases list.
Fixes #4710 

Additionally, this fixes a segfault that occurs when a user presses Enter while the "Recent databases" widget is in focus but no database is selected.

## Screenshots

clear-database

## Testing strategy

I did some manual tests.  
I'm new to this so let me know if/how I should implement tests.

## Type of change
- ✅ New feature (change that adds functionality)
- ✅ Bug fix (non-breaking change that fixes an issue)

### CODE REVIEW

File: src/gui/WelcomeWidget.cpp [hunk 1]
```diff
@@ -62,12 +62,26 @@ WelcomeWidget::~WelcomeWidget()
 
 void WelcomeWidget::openDatabaseFromFile(QListWidgetItem* item)
 {
-    if (item->text().isEmpty()) {
+    if (item == nullptr || item->text().isEmpty()) {
```
Comment (droidmonkey) [inline src/gui/WelcomeWidget.cpp:65]: `!item`

File: src/gui/WelcomeWidget.cpp [hunk 2]
```diff
@@ -62,12 +62,26 @@ WelcomeWidget::~WelcomeWidget()
...
         return;
     }
     emit openDatabaseFile(item->text());
 }
 
+void WelcomeWidget::removeFromLastDatabases(QListWidgetItem* item)
+{
+    if (item == nullptr || item->text().isEmpty()) {
```
Comment (droidmonkey) [inline src/gui/WelcomeWidget.cpp:73]: `!item`

File: src/gui/WelcomeWidget.cpp [hunk 3]
```diff
@@ -81,8 +95,12 @@ void WelcomeWidget::refreshLastDatabases()
 
 void WelcomeWidget::keyPressEvent(QKeyEvent* event)
 {
-    if (m_ui->recentListWidget->hasFocus() && (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter)) {
-        openDatabaseFromFile(m_ui->recentListWidget->currentItem());
+    if (m_ui->recentListWidget->hasFocus()) {
+        if (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter) {
+            openDatabaseFromFile(m_ui->recentListWidget->currentItem());
+        } else if (event->key() == Qt::Key_Delete) {
```
Comment (droidmonkey) [inline src/gui/WelcomeWidget.cpp:101]: Also add Qt::Key_Backspace

**Comment by phoerious** [comment]
I think there is still this bug that the field stays visible after clearing out all entries. That should be fixed as well.

**Comment by droidmonkey** [comment]
We can merge this into 2.6.0, it is a glaring UI issue

**Comment by JMcKiern** [comment]
Thanks :)

**Pull request merged**